backend/src/data/repositories/groups-to-permissions/groups-to-permissions.repository.ts

```diff
@@ -22,6 +22,32 @@ class GroupsToPermissions {
       permissionId,
     });
   }
+
+  public update(groupsToPermissions: {
+    groupId: number;
+    permissionIds: number[];
+  }): void {
+    const { groupId, permissionIds } = groupsToPermissions;
+
+    this.#GroupsToPermissionsModel
+      .query()
+      .where({ groupId })
+      .whereNotIn('permission_id', permissionIds)
+      .delete()
+      .execute();
```

Comment:
and this one is async, you need to await until groups deleted, and then await until everything is inserted or return Promise.all(permissionsIds.map())